account_reconciliation_widget_due_date/tests/test_account_reconciliation_widget_date_due.py

```diff
@@ -7,29 +7,40 @@
 
 
 class TestAccountReconciliationWidgetDueDate(TransactionCase):
-    def setUp(self):
-        super().setUp()
-        self.company = self.env.ref("base.main_company")
-        self.journal = self.env["account.journal"].create(
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(
+            context=dict(
+                cls.env.context,
+                mail_create_nolog=True,
+                mail_create_nosubscribe=True,
+                mail_notrack=True,
+                no_reset_password=True,
+                tracking_disable=True,
+            )
+        )
+        cls.company = cls.env.ref("base.main_company")
+        cls.journal = cls.env["account.journal"].create(
             {"name": "Test journal bank", "type": "bank", "code": "BANK-TEST"}
         )
-        self.account = self.env["account.account"].create(
+        cls.account = cls.env["account.account"].create(
             {
                 "name": "Account Receivable",
                 "code": "AR",
-                "user_type_id": self.env.ref("account.data_account_type_receivable").id,
+                "user_type_id": cls.env.ref("account.data_account_type_receivable").id,
                 "reconcile": True,
             }
         )
-        self.partner_a = self.env["res.partner"].create(
+        cls.partner_a = cls.env["res.partner"].create(
             {
                 "name": "Partner Test A",
-                "property_account_receivable_id": self.account.id,
+                "property_account_receivable_id": cls.account.id,
             }
         )
-        self.partner_b = self.partner_a.copy({"name": "Partner Test B"})
-        self.partner_c = self.partner_a.copy({"name": "Partner Test C"})
-        self.statement = self._create_account_bank_statement()
+        cls.partner_b = cls.partner_a.copy({"name": "Partner Test B"})
+        cls.partner_c = cls.partner_a.copy({"name": "Partner Test C"})
+        cls.statement = cls._create_account_bank_statement(cls)
```

Comment:
A better option here is to convert `_create_account_bank_statement` to `@classmethod`.